core/src/layers/foyer.rs

```diff
@@ -0,0 +1,207 @@
+use core::fmt;
+use std::{ops::Deref, sync::Arc};
+
+use crate::raw::*;
+use crate::*;
+use foyer::{Code, CodeError, HybridCache, HybridCacheBuilderPhaseStorage};
+use foyer_common::code::HashBuilder;
+use serde::{Deserialize, Serialize};
+
+pub struct FoyerLayer<S>
+where
+    S: HashBuilder + fmt::Debug,
+{
+    cache: Arc<HybridCache<CacheKey, Buffer, S>>,
+}
+
+impl<S> FoyerLayer<S>
+where
+    S: HashBuilder + fmt::Debug,
+{
+    pub async fn new(builder: HybridCacheBuilderPhaseStorage<CacheKey, Buffer, S>) -> Result<Self> {
```

Comment:
I believe we should accept a `HybridCache<String, Buffer>` directly.

The `HybridCacheBuilderPhaseStorage` API itself is part of builder, and should not be reply on directly.